program-test/src/lib.rs

```diff
@@ -698,3 +729,59 @@ impl ProgramTestBanksClientExt for BanksClient {
         ))
     }
 }
+
+struct DroppableTask<T>(Sender<()>, JoinHandle<T>);
+
+impl<T> Drop for DroppableTask<T> {
+    fn drop(&mut self) {
+        if let Err(e) = self.0.try_send(()) {
+            debug!("Error cancelling task: {:?}", e);
+        }
+    }
+}
+
+pub struct ProgramTestContext {
+    pub banks_client: BanksClient,
+    pub payer: Keypair,
+    pub last_blockhash: Hash,
+    _bank_task: DroppableTask<()>,
+}
+
+impl ProgramTestContext {
+    fn new(
+        bank_forks: Arc<RwLock<BankForks>>,
+        banks_client: BanksClient,
+        payer: Keypair,
+        last_blockhash: Hash,
+        genesis_config: GenesisConfig,
+    ) -> Self {
+        // Run a simulated PohService to provide the client with new blockhashes.  New blockhashes
+        // are required when sending multiple otherwise identical transactions in series from a
+        // test
+        let target_tick_duration = genesis_config.poh_config.target_tick_duration;
+        let (tx, mut rx) = mpsc::channel(1);
+        let bank_task = DroppableTask(
+            tx,
+            tokio::spawn(async move {
+                loop {
+                    if rx.try_recv().is_ok() {
```

Comment:
An AtomicBool is generally used in other parts of the code for this kind of exit pattern.   I'm not opposed to a channel if you want to keep it though.

Here's one example of using the atomic bool:
https://github.com/solana-labs/solana/blob/d6873b82ab61071eb87f4c18b2f40cd24d44bbc3/core/src/transaction_status_service.rs#L33

Reply:
That was actually my initial approach, but then I thought a message would be clearer.  I'll change it to an AtomicBool to keep things consistent in the codebase